src/contrib/mlir/pass/mlir_subgraph_extraction.cpp

```diff
@@ -254,7 +254,10 @@ bool MLIRSubgraphExtractionPass::run_on_function(std::shared_ptr<Function> func)
 
             for (descriptor::Input* in_desc : input_descs)
             {
-                in_desc->replace_output(ck, i);
+                if (std::find(nodes.begin(), nodes.end(), in_desc->get_node()) == nodes.end())
```

Comment:
I think `nodes` is an `unordered_set`. Could you please use nodes.find instead to use the hash implementation of find?